core/src/main/java/com/bakdata/dedupe/candidate_selection/online/OnlineCandidate.java

```diff
@@ -21,19 +21,36 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-package com.bakdata.deduplication.candidate_selection;
+package com.bakdata.dedupe.candidate_selection.online;
 
+import lombok.NonNull;
 import lombok.Value;
 
+/**
+ * Represents a candidate pair that was generated with an {@link OnlineCandidateSelection}.
+ *
+ * @param <T> the type of the records.
+ */
 @Value
-public class Candidate<T> {
+public class OnlineCandidate<T> implements com.bakdata.dedupe.candidate_selection.Candidate<T> {
+    /**
+     * The new record that triggered the {@link OnlineCandidateSelection}.
+     */
+    @NonNull
     T newRecord;
```

Comment:
rename fields maybe and then use lombok?

Reply:
The idea is that we have Online/Offline candidates with the same interface for clustering.
For offline, there is no inherent meaning of firstRecord and secondRecord. In Online, we can use newRecord and oldRecord with more semantics.
Not sure how you can make it more beautiful with lombok. Happy to take a patch.

Reply:
Ah, so both methods exist. I did not know that. Then it's fine